contracts/test/recovery-test.ts

```diff
@@ -248,4 +285,52 @@ describe("Recovery", async function () {
         .recoverWallet(addressSignature, hashAttacker, salt, wallet1Key),
     ).to.be.rejectedWith("VG: Signature not verified for wallet address");
   });
+
+  it("should NOT allow a bundle to be executed on a wallet with the same BLS pubkey but different address (replay attack)", async function () {
```

Comment:
I believe this covers `A bundle signed for a different address cannot be replayed onto a wallet with the same bls key` test case that was needed, but it is difficult to tell as this key switch via recovery will de-register the first wallet with that pubkey.